eth/core/chain_writer.go

```diff
@@ -50,36 +51,56 @@ func (bc *blockchain) WriteGenesisBlock(block *types.Block) error {
 	return err
 }
 
-// InsertBlockAndSetHead inserts a block into the blockchain and sets the head.
-func (bc *blockchain) InsertBlockAndSetHead(block *types.Block) error {
+// InsertBlockWithoutSetHead inserts a block into the blockchain without setting it as the head.
+func (bc *blockchain) InsertBlockWithoutSetHead(block *types.Block) error {
+	// Call the private method to insert the block without setting it as the head.
+	_, _, err := bc.insertBlockWithoutSetHead(block)
+	// Return any error that might have occurred.
+	return err
+}
+
+// insertBlockWithoutSetHead inserts a block into the blockchain without setting it as the head.
+func (bc *blockchain) insertBlockWithoutSetHead(
+	block *types.Block,
+) ([]*types.Receipt, []*types.Log, error) {
 	// Validate that we are about to insert a valid block.
+	// If the block number is greater than 1,
+	// it means it's not the genesis block and needs to be validated. TODO kinda hood.
 	if block.NumberU64() > 1 { // TODO DIAGNOSE
```

Comment:
What is the purpose of this conditional?

Reply:
a bug we can't figure out yet LOL